### ISSUE TITLE
question: Why use framebuffer and renderbuffer id rather than pointer in StateActiveObjects?

### ISSUE THREAD

**Issue by Michael-Lfx** [Think-Silicon/GLOVE#36]
There are some pointer members in `StateActiveObjects` such as `ShaderProgram *` and `Texture *`. But there are framebuffer and renderbuffer id also. Why not keep consistent? I have observed that things like `Context::GetIntegerv`, `Context::GetFloatv` and `Context::GetBooleanv` require GL object name aka id. However other operations such `Context::Bind*` will work better with pointer record in `StateActiveObjects`.

**In short, should we use std::tuple to save object name and pointer pair in `StateActiveObjects` or add an object name member to models like `Texture`, `BufferObject` etc. ?**

```
GLuint mActiveFramebufferObjectID
GLuint mActiveRenderbufferObjectID
```

https://github.com/Think-Silicon/GLOVE/blob/3ec146c99717cc302ce256fa20b03090bc6cb8ec/GLES/source/state/stateActiveObjects.h#L45

**Comment by abasilak** [comment]
> There are some pointer members in `StateActiveObjects` such as `ShaderProgram *` and `Texture *`. But there are framebuffer and renderbuffer id also. Why not keep consistent?

True. Ideally, the _ID_ is the key information that needs to be stored per active object.
We will solve this inconsistency in a future release version.
